src/content/subgames.h

```diff
@@ -53,25 +56,32 @@ struct SubgameSpec
 			const std::string &menuicon_path = "",
 			const std::string &author = "", int release = 0,
 			const std::string &first_mod = "",
-			const std::string &last_mod = "") :
+			const std::string &last_mod = "",
+			const std::unordered_set<std::string> &aliases = {}) :
 			id(id),
 			title(title), author(author), release(release),
 			first_mod(first_mod),
 			last_mod(last_mod),
 			path(path),
 			gamemods_path(gamemods_path),
 			addon_mods_paths(addon_mods_paths),
-			menuicon_path(menuicon_path)
+			menuicon_path(menuicon_path),
+			aliases(aliases)
 	{
 	}
 
 	bool isValid() const { return (!id.empty() && !path.empty()); }
 	void checkAndLog() const;
 };
 
+struct GameFindPath;
```

Comment:
since `GameFindPath` is only defined in `subgames.cpp` this should not be here

Reply:
It is here, because of `getAvailableGamePaths()` being declared a few lines later, and that returns a map containing this struct as value (which also has a `using` defined type directly below this line).

Reply:
I know that it is, but it fundamentally does not make sense to declare an opaque type like this in the *header* because no other source file can possible use it like this.

Reply:
So should I move the definition to the header?

Reply:
You don't need to expose the function at all, neither the struct. You could use forwards-declarations in the source file directly.